### DISCUSSION TITLE
Allow to have multiple DEBUG window instances

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#4671] [Editor]
It would play nice with multiple displays.

For example, using one for testing the game normally, and another to have custom GUI helper panels that assist testing and other dev tasks.

**_Edit:_**
It would be up to the user to share mouse or not.

Also there's a matter about DEBUG windows → they share the same game state or not?

Different game states per-window are useful for testing and compare side-by-side stuff like: networking game state sync, determinism, randomness and emergent behavior.

But also share game state would be useful: see things from different cameras, have GUI that shows data updated in real-time, etc.

_(Note: such discrepancies are because in Godot there's no real-time state sync Editor ←→ Game; so normal scripts for in-game state, and tool scripts for in-editor state)_

**Comment by Calinou** [comment]
This is already supported in 4.0.alpha, but it can't be backported to 3.x as it relies on a backwards-incompatible debugger rewrite.

![image](https://user-images.githubusercontent.com/180032/173852171-f5bcc009-62fc-4cb4-a28c-d6613668542a.png)

> Also there's a matter about DEBUG windows → they share the same game state or not?

No, each running project instance is a completely independent processes. Sharing the game state across different processes is outside the scope of the debugger. That said, you can synchronize the editor camera with the running project in both 2D and 3D already:

![image](https://user-images.githubusercontent.com/180032/173852501-7e3ea0c6-53a5-4bc6-876f-83dab9c4f912.png)

(The icon is no longer grayed out when a project is running.)

**Reply by azur-wolve** [reply]
Thanks for answer :)

Then if I've understood correctly, in **_Godot 4_** when running multiple instances, a completely different DEBUG is running.

Can also run the same instance, but with multiple windows? (independently of which Display, and `Viewport`|`Camera` each shows)
